tests/tools/fio/fio.go

```diff
@@ -5,45 +5,130 @@
+	FioExeEnvKey = "FIO_EXE"
+
+	// FioDockerImageEnvKey specifies the docker image tag to use. If
+	// FioExeEnvKey is set, the local executable will be used instead of
+	// docker, even if this variable is also set. The exception is if
+	// FioUseDockerEnvKey is not an empty string, which will force
+	// use of the fio docker image independent of FioExeEnvKey.
+	FioDockerImageEnvKey = "FIO_DOCKER_IMAGE"
+
+	// LocalFioDataPathEnvKey is the local path where fio data will be
+	// accessible. If not specified, defaults to the default temp directory (os.TempDir)
+	LocalFioDataPathEnvKey = "LOCAL_FIO_DATA_PATH"
+
+	// HostFioDataPathEnvKey specifies the path where fio data will be written,
+	// relative to the docker host. If left blank, defaults to local fio data path
+	// (works unless running via docker from within a container, e.g. for development)
+	HostFioDataPathEnvKey = "HOST_FIO_DATA_PATH"
+
+	// FioUseDockerEnvKey forces the fio runner to use the docker image, even if
+	// an executable path is provided.
+	FioUseDockerEnvKey = "FIO_USE_DOCKER"
+)
+
+// Known error messages
+var (
+	ErrEnvNotSet = fmt.Errorf("must set either %v or %v", FioExeEnvKey, FioDockerImageEnvKey)
+)
+
 // Runner is a helper for running fio commands
 type Runner struct {
-	Exe     string
-	DataDir string
-	Global  Config
+	Exe             string
+	ExecArgs        []string
+	LocalDataDir    string
+	FioWriteBaseDir string
+	Global          Config
 }
 
 // NewRunner creates a new fio runner
-func NewRunner() (*Runner, error) {
-	Exe := os.Getenv("FIO_EXE")
-	if Exe == "" {
-		Exe = "fio"
-	}
+func NewRunner() (fr *Runner, err error) {
+	exeStr := os.Getenv(FioExeEnvKey)
+	imgStr := os.Getenv(FioDockerImageEnvKey)
+	localFioDataPathStr := os.Getenv(LocalFioDataPathEnvKey)
+	hostFioDataPathStr := os.Getenv(HostFioDataPathEnvKey)
+	forceDocker := os.Getenv(FioUseDockerEnvKey) != ""
+
+	var exeArgs []string
 
-	dataDir, err := ioutil.TempDir("", "fio-data")
+	var fioWriteBaseDir string
+
+	var Exe string
+
+	dataDir, err := ioutil.TempDir(localFioDataPathStr, "fio-data-")
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to create temp directory for fio runner")
+	}
+
+	switch {
+	case exeStr != "" && !forceDocker:
+		// Provided a local FIO executable to run
+		Exe = exeStr
+
+		fioWriteBaseDir = dataDir
+
+	case imgStr != "":
+		// Provided a docker image to run inside
+		Exe = dockerExe
+
+		dataDirParent, dataDirName := filepath.Split(dataDir)
+		fioWriteBaseDir = filepath.Join(fioDataContainerPath, dataDirName)
+
+		// If the host path wasn't provided, assume it's the same as the local
+		// data directory path and we are not running from within a container already
+		if hostFioDataPathStr == "" {
+			hostFioDataPathStr = dataDirParent
+		}
+
+		exeArgs = []string{
+			"run",
+			"--rm",
+			"-v",
+			fmt.Sprintf("%s:%s", hostFioDataPathStr, fioDataContainerPath), // /c/Users/usr/fio-data:/fio-data
```

Comment:
Is the comment at the end needed?
```suggestion
			fmt.Sprintf("%s:%s", hostFioDataPathStr, fioDataContainerPath),
```